nil/cmd/nil_block_generator/internal/commands/client_test.go

```diff
@@ -0,0 +1,121 @@
+package commands
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"testing"
+
+	"github.com/NilFoundation/nil/nil/common/logging"
+	"github.com/NilFoundation/nil/nil/internal/collate"
+	"github.com/NilFoundation/nil/nil/services/nilservice"
+	"github.com/NilFoundation/nil/nil/tests"
+	"github.com/NilFoundation/nil/nil/tools/solc"
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/suite"
+)
+
+const contract_str = `pragma solidity ^0.8.0;
+	contract SimpleStorage {
+	    uint256 private value;
+		event ValueChanged(uint256 newValue);
+	    function increment() public {
+	        value += 1;
+	        emit ValueChanged(value);
+	    }
+	}`
+
+type NilBlockGeneratorTestSuite struct {
+	tests.RpcSuite
+	contractPath string
+	contractName string
+	method       string
+	logger       zerolog.Logger
+}
+
+func CompileContract(path, name, basePath string) error {
+	contracts, err := solc.CompileSource(path)
+	if err != nil {
+		return err
+	}
+
+	// write bytecode to file
+	bytecodePath := basePath + ".bin"
+	err = os.WriteFile(bytecodePath, []byte(contracts[name].Code), fileMode)
+	if err != nil {
+		return err
+	}
+
+	// write ABI to file
+	data, err := json.Marshal(contracts[name].Info.AbiDefinition)
+	if err != nil {
+		return err
+	}
+	abiPath := basePath + ".abi"
+	err = os.WriteFile(abiPath, data, fileMode)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *NilBlockGeneratorTestSuite) SetupTest() {
+	err := CleanFiles()
+	s.Require().NoError(err)
+
+	s.logger = logging.NewLogger("test_nil_block_generator")
+
+	// prepare compiled contract
+	s.contractPath = "TestContract"
+	s.contractName = "SimpleStorage"
+	s.method = "increment"
+	contractCodePath := s.contractPath + ".sol"
+	err = os.WriteFile(contractCodePath, []byte(contract_str), fileMode)
+	s.Require().NoError(err)
+	err = CompileContract(contractCodePath, s.contractName, s.contractPath)
+	s.Require().NoError(err)
+
+	nilserviceCfg := &nilservice.Config{
+		NShards:              2,
+		RPCPort:              defaultRPCPort,
+		Topology:             collate.TrivialShardTopologyId,
+		CollatorTickPeriodMs: 100,
+		GasBasePrice:         10,
+	}
+
+	s.Context, s.CtxCancel = context.WithCancel(context.Background())
+	s.SetT(s.T())
```

Comment:
You don't need it because you inherit your suite from `tests.RpcSuite`.

Reply:
thanks!, done